File: Python/Bronze/2017-02/WhyDidTheCowCrossTheRoadIII.py
#
#  USACO 2017 February - Bronze - Problem 3 - Why Did the Cow Cross the Road III
#
import sys

sys.stdin = open('cowqueue.in', 'r')
sys.stdout = open('cowqueue.out', 'w')

N = int(input())
cows = []

for _ in range(N):
	arr, dur = map(int, input().split())
	cows.append((arr, dur))

cows.sort()

cur_time = 0
for arr, dur in cows:
	if cur_time < arr:
		cur_time = arr
	
	cur_time += dur

print(cur_time)

File: CPlusPlus/Bronze/2017-02/WhyDidTheCowCrossTheRoadIII.cpp
/**
 *  USACO 2017 February - Bronze - Problem 3 - Why Did the Cow Cross the Road III
 */

#include <bits/stdc++.h>

using namespace std;

int main()
{
    freopen("cowqueue.in", "r", stdin);
    freopen("cowqueue.out", "w", stdout);

    int N;
    cin >> N;

    vector<pair<int, int>> cows(N);
    for (int i = 0; i < N; i++)
    {
        cin >> cows[i].first >> cows[i].second;
    }

    sort(cows.begin(), cows.end());

    int curTime = 0;

    for (int i = 0; i < N; i++)
    {
        curTime = max(curTime, cows[i].first) + cows[i].second;
    }

    cout << curTime << endl;

    return 0;
}